highlights.lua

```diff
@@ -4,9 +4,164 @@ local ts = require 'libraries.tree_sitter'
 
 local M = {}
 
-local function localPath()
-	local str = debug.getinfo(2, 'S').source:sub(2)
-	return str:match '(.*[/\\])'
+local function predicatesFor(doc)
+	local function getSource(n)
+		local startPt = n:start_point()
+		local endPt   = n:end_point()
+		local startRow, startCol = startPt:row() + 1, startPt:column() + 1
+		local endRow, endCol     = endPt:row() + 1, endPt:column()
+
+		return doc:get_text(startRow, startCol, endRow, endCol)
+	end
+
+	local function coerceToStr(n)
+		if type(n) ~= 'string' then
+			return getSource(n:one_node())
+		else
+			return n
+		end
+	end
+
+	local predicates = {
+		['eq?'] = function(ns, m)
+			local str = coerceToStr(m)
+
+			for _, n in ipairs(ns:nodes()) do
+				if getSource(n) ~= str then return false end
+			end
+			return true
+		end,
+
+		['any-eq?'] = function(ns, m)
```

Comment:
this is a real predicate? used by real nvim-treesitter queries? isnt it just an inverse of eq? am i stupid or is this stupid?

Reply:
When the capture consists of multiple nodes, `pred?` checks if each of the individual nodes satisfies a condition. `any-pred?` checks if any one of the nodes satisfies the condition.

Reply:
See https://neovim.io/doc/user/treesitter.html#lua-treesitter-all-predicate